onnxruntime/core/platform/windows/env.cc

```diff
@@ -148,28 +171,141 @@ class WindowsEnv : public Env {
+      LOGS_DEFAULT(ERROR) << "Processor <from> or <to> crossed group boundary";
+      return {-1,0};
+    }
+    return {group_id, processor_mask};
+  }
+
+  std::pair<KAFFINITY, KAFFINITY> GetGroupAffinity(const std::vector<std::string>& processor_id_strs) {
+    if (processor_id_strs.empty()) {
+      return {-1, 0};
+    }
+    std::set<int> processor_ids;
+    std::for_each(processor_id_strs.begin(),
+                  processor_id_strs.end(), [&](const std::string& processor_id_str) {
+                    processor_ids.insert( std::stoi(processor_id_str.c_str()));
+                  });
+    int processor_id = 1;
+    int processor_count = 0;
+    KAFFINITY group_id = 0;
+    KAFFINITY processor_mask = 0;
+    for (const auto& group_info : this->group_info_vec_) {
+      for (int i = 0; i < group_info.ActiveProcessorCount; ++i, ++processor_id) {
+        if (processor_ids.count(processor_id)) {
+          processor_mask |= 1ULL << i;
+          processor_count += 1;
+        }
+      }
+      if (processor_mask) {
+        break;
+      }
+      group_id++;
+    }
+    if (processor_count < processor_ids.size()) {
+      LOGS_DEFAULT(ERROR) << "Processor id(s) crossed group boundary";
+      return {-1, 0};
+    }
+    return {group_id, processor_mask};
+  }
+
+  std::vector<size_t> ReadThreadAffinityConfig(const std::string& affinity_str) {
+    try {
+      auto config_all_threads = SplitStr(affinity_str, ';');
+      std::vector<KAFFINITY> affinities;
+      for (const auto& config_per_threads : config_all_threads) {
+        if (config_per_threads.empty()) {
+          LOGS_DEFAULT(ERROR) << "Found empty affinity string!";
+          return {};
+        }
+        auto partition_by_hyphen = SplitStr(config_per_threads, '-');
+        if (partition_by_hyphen.size() == 2) {
+          int from = stoi(partition_by_hyphen[0]);
+          int to = stoi(partition_by_hyphen[1]);
+          auto group_affinity = GetGroupAffinity(from, to);
+          if (group_affinity.first == -1) {
+            return {};
+          }
+          affinities.push_back(group_affinity.first);
+          affinities.push_back(group_affinity.second);
+        } else {
+          auto partition_by_comma = SplitStr(config_per_threads, ',');
+          if (partition_by_comma.empty()) {
+            LOGS_DEFAULT(ERROR) << "Wrong affinity string format: " << config_per_threads;
+            return {};
+          } else {
+            auto group_affinity = GetGroupAffinity(partition_by_comma);
+            if (group_affinity.first == -1) {
+              return {};
+            }
+            affinities.push_back(group_affinity.first);
+            affinities.push_back(group_affinity.second);
+          }
+        }
+      }
+      size_t num_of_size_t = affinities.size() * sizeof(KAFFINITY) / sizeof(size_t);
+      std::vector<size_t> ret(num_of_size_t, 0);
+      memcpy(ret.data(), affinities.data(), num_of_size_t * sizeof(size_t));
```

Comment:
Will this work if affinities is an empty vector?

Reply:
yes - when affinity is empty nothing will be copied to output vector. 
In addition, I added another condition beforehand to deal with the empty case, pls see the latest commit.